OS X Build Errors

A new version of XCode is outputting a new warning, resulting in build errors.

```
steem/libraries/plugins/apis/account_history_api/account_history_api.cpp:159:17: error: lambda capture 'this' is not used [-Werror,-Wunused-lambda-capture]
      [&result, this](unsigned int sequence, const account_history_rocksdb::rocksdb_operation_object& op)
```

--- a/libraries/plugins/apis/account_history_api/account_history_api.cpp
+++ b/libraries/plugins/apis/account_history_api/account_history_api.cpp
@@ -156,7 +156,7 @@ DEFINE_API_IMPL( account_history_api_rocksdb_impl, get_account_history )
    get_account_history_return result;
 
    _dataSource.find_account_history_data(args.account, args.start, args.limit,
-      [&result, this](unsigned int sequence, const account_history_rocksdb::rocksdb_operation_object& op)
+      [&result](unsigned int sequence, const account_history_rocksdb::rocksdb_operation_object& op)
       {
          result.history[sequence] = api_operation_object( op );
       });

